File: src/types/feasibility.ts
export type FeasibilityLabel = 'LOW' | 'MODERATE' | 'GOOD' | 'STRONG' | 'VERY_STRONG'
export type ComponentCode = 'LOCAL_DEMAND' | 'COMPETITION_OPPORTUNITY' | 'SKILL_MATCH' | 'FINANCIAL_FIT' | 'MARKET_ACCESS' | 'SUPPLY_CHAIN'
export interface ScoreComponent { code: ComponentCode; title: string; score: number; weight: string; weightedContribution: string }
export interface ExplanationItem { code: string; title: string; explanation: string }
export interface GapItem { code: string; severity: 'LOW' | 'MEDIUM' | 'HIGH'; explanation: string }
export interface SkillEvidence { matchedRequiredSkills: string[]; missingRequiredSkills: string[]; matchedPreferredSkills: string[]; skillMatchScore: number }
export interface ResourceEvidence { availableRequiredResources: string[]; missingRequiredResources: string[]; availableOptionalResources: string[]; resourceReadinessScore: number }
export interface FeasibilityResult {
  rank: number; businessId: string; businessSlug: string; businessName: string; businessCategory: string; shortDescription: string
  finalFeasibilityScore: number; feasibilityLabel: FeasibilityLabel; isCurrentBusiness: boolean; components: ScoreComponent[]
  skills: SkillEvidence; resources: ResourceEvidence; capitalFitExplanation: string; financingMayBeRequired: boolean
  strengths: ExplanationItem[]; gaps: GapItem[]; nextActions: string[]; marketThreats: Array<{ code: string; explanation: string }>
}
export interface FeasibilityAnalysis { id: string; createdAt: string; analysisVersion: string; marketAnalysisId: string; marketAnalysisVersion: string; marketDataVersion: string; disclaimer: string; results: FeasibilityResult[]; topMatches: FeasibilityResult[] }

File: backend/app/schemas/feasibility.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class FeasibilityRequest(BaseModel):
    market_analysis_id: Optional[uuid.UUID] = None


class ScoreComponent(BaseModel):
    code: str
    title: str
    score: int
    weight: Decimal
    weighted_contribution: Decimal


class ExplanationItem(BaseModel):
    code: str
    title: str
    explanation: str


class GapItem(BaseModel):
    code: str
    severity: str
    explanation: str


class SkillEvidence(BaseModel):
    matched_required_skills: List[str]
    missing_required_skills: List[str]
    matched_preferred_skills: List[str]
    skill_match_score: int


class ResourceEvidence(BaseModel):
    available_required_resources: List[str]
    missing_required_resources: List[str]
    available_optional_resources: List[str]
    resource_readiness_score: int


class FeasibilityResult(BaseModel):
    rank: int
    business_id: uuid.UUID
    business_slug: str
    business_name: str
    business_category: str
    short_description: str
    final_feasibility_score: int
    feasibility_label: str
    is_current_business: bool
    components: List[ScoreComponent]
    skills: SkillEvidence
    resources: ResourceEvidence
    capital_fit_explanation: str
    financing_may_be_required: bool
    strengths: List[ExplanationItem]
    gaps: List[GapItem]
    next_actions: List[str]
    market_threats: List[dict]


class FeasibilityResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    created_at: datetime
    analysis_version: str
    market_analysis_id: uuid.UUID
    market_analysis_version: str
    market_data_version: str
    disclaimer: str
    results: List[FeasibilityResult]
    top_matches: List[FeasibilityResult]
